Userland/Libraries/LibAudio/Encoder.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Error.h>
+#include <AK/Span.h>
+#include <LibAudio/Sample.h>
+
+namespace Audio {
+
+class Encoder {
+public:
+    virtual ~Encoder() = default;
+
+    virtual ErrorOr<void> write_samples(Span<Sample> samples) = 0;
+
+    virtual ErrorOr<void> finalize() = 0;
```

Comment:
Can you add some comments to be very explicit about the role of each method?
That's just as a prevention measure, hope it can help all future encoders to be consistent.